### ISSUE TITLE
Relation Widget resets when using autocomplete 

### ISSUE THREAD

**Issue by haneatic** [netlify/netlify-cms#4061]
**Description**
If you use the autocomplete functionality of the relation widget together with the multiple option it resets the selected items completely and only keeps the last one you picked.

**To Reproduce**
1. Open the dropdown menu of the relation widget
2. Pick any number of items
3. Search for any non selected item
4. Select an entry you found through the search
5. It should be resetted

**Expected behavior**
It should keep previous selected items.

**CMS configuration**
```yml
backend:
  name: git-gateway
  branch: master

local_backend: true

site_url: https://example.com

# publish_mode: editorial_workflow
media_folder: assets/uploads

collections: # A list of collections the CMS should be able to edit
  - name: 'Itemlist'
    label: 'Itemlist'
    create: true
    folder: '_itemlist'
    fields:
      - label: 'Title'
        name: 'title'
        widget: 'string'
        required: true
      - label: 'Items'
        name: 'Items'
        widget: 'list'
        fields:
          - label: 'Name'
            name: 'Name'
            widget: 'string'
            required: true

  - name: 'Collector'
    label: 'Collector'
    create: true
    folder: '_collector'
    fields:
      - label: 'Title'
        name: 'title'
        widget: 'string'
        required: true
      - label: 'Items'
        name: 'items'
        widget: 'relation'
        collection: 'Itemlist'
        valueField: 'Items.*.Name'
        searchFields: ['Items.*.Name']
        multiple: true
```

**Additional context**
I actually tried to track down the issue and I think that [this](https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-widget-relation/src/RelationControl.js#L52) line is either part of it or responsible for it. When selecting an item you searched for it replaces the already selected items with `undefined`

**Comment by haneatic** [comment]
If the map function from L52 gets removed completely, it works fine. The issue comes from using the search function because if you use the search function it filters the options which are provided to the function `getSelectedValue`. The map function then can not find every selected value inside of the options array anymore since they are not holding *all* possible options but only the *currently* possible options which fit the entered search term. As a result it replaces previous selected values with `undefined` (in case they didn't fit the search term) and it removes them from the selection. I don't know yet if it's smart to remove the map function completely since I assume it's there for a reason.

**Comment by haneatic** [comment]
Before [this](https://github.com/netlify/netlify-cms/commit/c7e0fe8492d09a3d151c608f50da844f421362ed) commit it worked fine.

**Comment by erezrokah** [comment]
@haneatic posted a fix for this issue, thanks for reporting.
See https://github.com/netlify/netlify-cms/pull/4063
If you'd like to test it you can load the CMS via:
https://deploy-preview-4063--cms-demo.netlify.app/dist/netlify-cms.js

**Comment by falconmasters** [comment]
I have the same problem
@erezrokah I am a little bit lost, could you please tell me how to apply your fix to my proyect?

**Comment by haneatic** [comment]
@falconmasters depends on which framework you use. If you use GatsbyJS for example you can create a custom index.html file in `static/admin`, see https://www.netlifycms.org/docs/add-to-your-site/

You only have to exchange the url to the productive version of NetlifyCMS to @erezrokah links. It works fine then. 

@erezrokah thanks for your help!

**Comment by erezrokah** [comment]
Thanks @haneatic  and everyone. We'll try to get this released tomorrow.

**Comment by falconmasters** [comment]
Once released how do I update my site?
I am using Gatsby

**Comment by erezrokah** [comment]
If you're using `unpkg`, you should get it automatically.
If using the CMS via `package.json`, you'll need to update the CMS dependency (using `npm` or `yarn`, etc)

**Comment by erezrokah** [comment]
This is released in `netlify-cms@2.10.55` and `netlify-cms-app@2.12.19`
